pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -178,3 +181,101 @@ func (i *deploymentClientImpl) listDeploymentsForConfig(namespace, configName st
+
+	details := ""
+	allDetails := []string{}
+	willHaveImage, hasImageChangeTrigger := false, false
+	if config.Details != nil && len(config.Details.Message) > 0 {
+		// Populate details with the previous message so that in case we stumble upon
+		// an unexpected client error, the message won't be overwritten
+		details = config.Details.Message
+	}
+	// Look into image change triggers and find out possible deployment failures such as
+	// missing image stream tags with or without build configurations pointing at them
+	for _, trigger := range config.Triggers {
+		if trigger.Type != deployapi.DeploymentTriggerOnImageChange || trigger.ImageChangeParams == nil || !trigger.ImageChangeParams.Automatic {
+			continue
+		}
+		hasImageChangeTrigger = true
+		name := trigger.ImageChangeParams.From.Name
+		tag := trigger.ImageChangeParams.Tag
+		istag := imageapi.JoinImageStreamTag(name, tag)
+
+		// Check if the image stream tag pointed by the trigger exists
+		if _, err := c.osClient.ImageStreamTags(config.Namespace).Get(name, tag); err != nil {
+			if !errors.IsNotFound(err) {
+				glog.V(2).Infof("Error while trying to get image stream tag %q: %v", istag, err)
+				return details, existingDeployments, latestDeploymentExists, nil
+			}
+			// In case the image stream tag was not found, then it either doesn't exist or doesn't exist yet
+			// (a build configuration output points to it so it's going to be populated at some point in the
+			// future)
+			details = fmt.Sprintf("The image trigger for istag/%s will have no effect because istag/%s does not exist.", istag, istag)
+			bcList, err := c.osClient.BuildConfigs(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
+			if err != nil {
+				glog.V(2).Infof("Error while trying to list buildConfigs: %v", err)
+				return details, existingDeployments, latestDeploymentExists, nil
+			}
+
+			for _, bc := range bcList.Items {
+				if bc.Spec.Output.To != nil && bc.Spec.Output.To.Kind == "ImageStreamTag" {
+					parts := strings.Split(bc.Spec.Output.To.Name, ":")
+					if len(parts) != 2 {
+						glog.V(2).Infof("Invalid image stream tag: %q", bc.Spec.Output.To.Name)
+						return details, existingDeployments, latestDeploymentExists, nil
+					}
+					if parts[0] == name && parts[1] == tag {
+						details = fmt.Sprintf("The image trigger for istag/%s will have no effect because istag/%s does not exist. If istag/%s is expected, check bc/%s which produces istag/%s.", istag, istag, istag, bc.Name, istag)
+						break
+					}
+				}
+			}
+			// Try to see if the image stream exists, if not then the build will never be able to update the
+			// tag in question
+			if _, err := c.osClient.ImageStreams(config.Namespace).Get(name); err != nil {
+				glog.V(2).Infof("Error while trying to get image stream %q: %v", name, err)
+				if errors.IsNotFound(err) {
+					details = fmt.Sprintf("The image trigger for istag/%s will have no effect because is/%s does not exist.", istag, name)
+				}
```

Comment:
Don't use "istag/" syntax in messages from a controller.  Say "image stream tag foo:bar"